apps/token/lib/token/transaction/transaction.ex

```diff
@@ -0,0 +1,50 @@
+defmodule Token.Transaction do
+  @moduledoc """
+  Structure that represents a transaction and functions to work on transactions
+  """
+
+  alias Token.{Wallet, Transaction, Crypto}
+
+  defstruct [
+    :hash,
+    :inputs, # a list of list of the form [[previous_tx_hash, output_index]]
+    :public_key,
+    :signature,
+    :outputs # a list of list of the form [[recipient, value]]
+  ]
+
+  def new_transaction(%Wallet{} = wallet, inputs, outputs) do
+    tx = %Transaction{
+      inputs: inputs,
+      public_key: wallet.public_key,
+      outputs: outputs
+    }
+    sig =
+      tx
+      |> signing_string()
+      |> Wallet.sign(wallet)
+    signed_tx = %{tx | signature: sig}
+    %{signed_tx | hash: compute_hash(signed_tx)}
+  end
+
+  def new_coinbase_transaction(outputs) do
+    tx = %Transaction{
+      outputs: outputs,
+      inputs: [["0", 0]] # coinbase transaction not validated
```

Comment:
`coinbase` eh?

Reply:
yes, I may have misused the term, but it's from: https://bitcoin.org/en/glossary/coinbase-transaction
Basically, a transaction that generates tokens